client/src/components/formHelpers/form-fields.tsx

```diff
@@ -59,13 +59,22 @@ function FormFields(props: FormFieldsProps): JSX.Element {
         validationError = (err as { message?: string })?.message;
       }
     }
-    const validationWarning = composeValidators(
-      name === 'githubLink' || isEditorLinkAllowed ? null : editorValidator,
-      fCCValidator,
-      httpValidator,
-      isLocalLinkAllowed ? null : localhostValidator,
-      pathValidator
-    )(value);
+
+    // If editor link is not allowed, use editor validator
+    // If local link is not allowed, use local validator
+    // If local link is allowed, use path validator
+    // Always used fCCValidator and httpValidator
+    const validators = [fCCValidator, httpValidator];
+    if (name !== 'githubLink' && !isEditorLinkAllowed) {
+      validators.push(editorValidator);
+      if (isLocalLinkAllowed) {
+        validators.push(pathValidator);
+      }
```

Comment:
Is this intentionally nested?

Reply:
Unfortunately 🙃 
This is a hacky way to say: _"This challenge input is not a GitHub (code-link) submission, and is not a certification project"_